src/VersionTool/UpdatePatchCommand.cs

```diff
@@ -0,0 +1,273 @@
+                    if (repo != null)
+                    {
+                        // update version
+                        UpdateVersionCommand.UpdateVersion(
+                            new List<string>(new[] { Path.Combine(repo.Path, "src", packageName) }),
+                            new List<string>(),
+                            parsedPackageUpdate[1]);
+
+                        // update entry in packages
+                        var package = patchConfig.Packages.First(p => string.Equals(p.Name, packageName, StringComparison.OrdinalIgnoreCase));
+                        package.NewVersion = newPackageVersion;
+                    }
+                    else
+                    {
+                        // update dependency
+                        UpdateDependencyCommand.UpdateDependency(
+                            new List<string>(new[] { directory }),
+                            new List<string>(),
+                            parsedPackageUpdate[0],
+                            parsedPackageUpdate[1]);
+                    }
+                }
+            }
+
+            // Handle repo updates
+            if (updateRepoOption.HasValue())
+            {
+                var repo = patchConfig.Repos.Where(r => string.Equals(r.Name, updateRepoOption.Value(), StringComparison.OrdinalIgnoreCase)).FirstOrDefault();
+
+                if (repo == null)
+                {
+                    throw new InvalidOperationException($"Repo {updateRepoOption.Value()} not found in patch config.");
+                }
+
+                UpdatePackagesInRepo(repo, directory, patchConfig);
+            }
+
+            // Cascade updates
+            foreach (var repo in patchConfig.Repos.OrderBy(r => r.Order))
+            {
+                if (ContainsModifiedSrcProjectFile(repo.Path))
+                {
+                    UpdatePackagesInRepo(repo, directory, patchConfig);
+                }
+            }
+
+            // Update patch config with updated packages
+            UpdatePatchConfig(patchConfig, patchConfigPath, repoConverter);
+
+            return 0;
+        }
+
+        private static void UpdatePackagesInRepo(Repository repo, string directory, PatchConfig patchConfig)
+        {
+            foreach (var package in Utilities.EnumerateProjects(new[] { Path.Combine(repo.Path, "src") }))
+            {
+                var unUpdatedPackage = patchConfig.Packages.Where(p => p.Name == package.Name && p.CurrentVersion == p.NewVersion).FirstOrDefault();
+                if (unUpdatedPackage != null)
+                {
+                    Console.WriteLine($"{package.Name} is a src project and has not been updated in this patch, updating version.");
+
+                    var rule = GetRuleForPackage(package, patchConfig.Rules);
+
+                    if (rule == null)
+                    {
+                        throw new Exception($"No update rule found for package {package.Name} {package.Version}.");
```

Comment:
Could you include an example in the error so somebody that runs it has an idea on what they are expected to do?